src/appengine/libs/issue_management/github.py

```
@@ -0,0 +1,177 @@
+  testcase.repo_id = repo.id
+  testcase.issue_num = issue.number
+
+
+def file_issue(testcase):
+  """File an issue to the GitHub repo of the project"""
+  if not filing_enabled(testcase):
+    return
+
+  access = get_access()
+  if not access:
+    logs.log_error("Unable to access GitHub account and file the issue.")
+    return
+  repo = get_repo(testcase, access)
+  if not repo:
+    logs.log_error("Unable to locate GitHub repository and file the issue.")
+    return
+  issue = post_issue(repo, testcase)
+  update_testcase_properties(testcase, repo, issue)
+
+
+def issue_recorded(testcase):
+  """Verify the issue has been filed."""
+  return hasattr(testcase, 'github_repo_id') \
+      and testcase.github_repo_id is not None \
+      and hasattr(testcase, 'github_issue_num') \
+      and testcase.github_issue_num is not None
+
+
+def get_issue(testcase, access):
+  """Locate the issue of the testcase."""
+  repo_id = testcase.github_repo_id
+  issue_num = testcase.github_issue_num
+  try:
+    repo = access.get_repo(repo_id)
+  except github.UnknownObjectException:
+    logs.log_error("Unable to locate the GitHub repository " f"id {repo_id}.")
+    return None
+
+  try:
+    target_issue = repo.get_issue(issue_num)
+  except github.UnknownObjectException:
+    logs.log_error("Unable to locate the GitHub issue " f"number {issue_num}.")
+    target_issue = None
+  return target_issue
+
+
+def close_issue_with_comment(testcase, issue):
+  """Generate closing comment, comment, and close the GitHub issue."""
+  issue_close_comment = get_issue_close_comment(testcase)
+  issue.create_comment(issue_close_comment)
+  issue.edit(state='closed')
+
+
+def close_issue(testcase):
+  """Close the issue on GitHub, when the same issue is closed on Monorail."""
+  if not issue_recorded(testcase):
+    return
+  access = get_access()
+  if not access:
+    logs.log_error("Unable to access GitHub account and close the issue.")
```

Let's raise a RuntimeError exception instead, rather than logging an error and exiting. This is a pretty bad thing to happen.

---

No need to handle it. It should bubble up as a configuration error that the infra admins (i.e. us) have to fix.